05.ls/ls.rb

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env ruby
+require 'etc'
+require 'time'
+require 'optparse'
+
+def main
+  options = ARGV.getopts('a', 'l', 'r')
+
+  # 上からそれぞれのオプションの処理を実行、なければオプション無しの処理
+  files = if options['a']
+            input_list('a')
+          else
+            input_list('no')
+          end
+
+  files = reverse_the_array(files) if options['r']
+
+  if options['l']
+    output_list_in_long_format(files)
+  else
+    output_list(files)
+  end
+end
+
+# ファイルのパスを配列にいれる
+def input_list(arg)
+  files = []
+  current_directory = Dir.pwd
+  if arg == 'a'
+    Dir.glob('*', File::FNM_DOTMATCH) do |file|
+      files << File.join(current_directory, file)
+    end
+  else
+    Dir.glob('*') do |file|
+      files << File.join(current_directory, file)
+    end
+  end
```

Comment:
オプションだけわけてもいいかもですね
```suggestion
  option = arg == 'a' ? File::FNM_DOTMATCH : nil
  Dir.glob('*', option) do |file|
    files << File.join(current_directory, file)
  end
```
みたいな感じで。

Reply:
optionにnilが渡される場合にTypeErrorになるのですがどうしたらいいですか？

Reply:
@dunonoshiwu すいません、これデフォルトnilじゃなくて0みたいですね。
https://docs.ruby-lang.org/ja/latest/method/Dir/s/=5b=5d.html
`option = arg == 'a' ? File::FNM_DOTMATCH : 0` にしたらどうでしょうか？

Reply:
0にしたら動作しました！！自分は第2引数に何を渡せばいいのかわからなかったのでめちゃ感動です！

Reply:
@dunonoshiwu 一応解説しておくとリファレンスマニュアルに
`glob(pattern, flags = 0, base: nil, sort: true) -> [String]` みたいな記述があると思います。
https://docs.ruby-lang.org/ja/latest/method/Dir/s/=5b=5d.html
この第2引数が該当のところで `flag = 0` で、 `= 0` のところが `何もわたさなかったら0がデフォルト値として設定される` という意味になります！